Makefile

```diff
@@ -0,0 +1,29 @@
+ifdef MVN_FLAGS
+    $(info MVN_FLAGS defined manually)
+else
+    MVN_FLAGS := "-e"
+endif
+
+
+.SILENT:
+
+build: build-parent build-core build-cli
```

Comment:
Would it not be easier to just build the whole project instead of the parent followed by all modules?

Reply:
Also relates to the use of a make file I guess

Reply:
how would you build the whole project? I find myselfs in hot water with maven and composite projects over and over again. Means

- i build the project using `mvn --also-make -pl cli clean package` (first build, all fine)
- then i do changes on core
- run `mvn --also-make -pl cli clean package`

And what usually happens is that mvn will not rebuild core OR the build of cli will not include the last artifact. Those issues with mvn and composite is one of the main reasons i use gradle, because this is entirely handled there (you consume composite builds using projects artifacts ... not pseudo coordinates with a magic version number that matches some local `0-SNAPSHOT`.

So IMHO, i use a clean rebuild of each module every time in an manual, orchestrated way, so that mvn will not screw up. This ensure i always test what i intend to.

If you have a better idea or a tip, a magic flag to fix this, please share it with me - but that above is basically my go-to with mvn projects as "better safe then outdated"

Reply:
> how would you build the whole project? 

Basically you call `mvn clean package` and it will build the main project and all modules. You would use a flag like `-pl` to only build a specific module

> And what usually happens is that mvn will not rebuild core OR the build of cli will not include the last artifact.

If you do a `clean package` then the artifact is always updated, if you just do `package` then it can happen that artifact is not completely up to date, all though I have only seen this happen when I save file mid build.

> If you have a better idea or a tip, a magic flag to fix this, please share it with me

On CI systems I always do a `clean install` make sure no old build files are used and that the actifacts are installed on the local m2 repo for later use.
Locally it depends on what I need, just a compile to debug some tests I just call `compile` otherwise its mostly either `install` or `clean install`

Reply:
thanks for your insight. i adjusted the task 

```
build-all:
	./mvnw $(MVN_FLAGS) -DskipTests clean install
```

so when building the entire project, it will use you recommendation. There is still build-parent, build-core,build-cli for specific builds, but the CI and the readme generally use `make build-all`

Does this cover your suggestion or would there be more?

Reply:
Yeah that covers it